Code_Saeyeon_review_par_Cedric.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Aug  1 16:29:27 2022
+
+@author: Saeyeon
+"""
+
```

Comment:
General comments.

Code looks correct to me on first read (yay!).

Code organization could be improved both to improve clarity (make it easier to understand) and to avoid code duplication (the same variables/array coefficients are accessed several times to perform essentially the same three operations in the case of the original data, shuffling keeping pairs and full shuffling).
[Let's talk in person what I would recommend in terms of code structure]